src/components/elements/input/TimeDurationInput.tsx

```diff
@@ -0,0 +1,65 @@
+import { Typography } from '@mui/material';
+import { Stack } from '@mui/system';
+import { memoize } from 'lodash-es';
+import { useMemo } from 'react';
+import NumberInput from '@/components/elements/input/NumberInput';
+import { TextInputProps } from '@/components/elements/input/TextInput';
+
+type Props = { value?: number; onChange?: (val: number | null) => void } & Omit<
+  TextInputProps,
+  'value' | 'onChange'
+>;
+
+const TimeDurationInput = ({ value, onChange, label, ...props }: Props) => {
+  const [hours, minutes] = useMemo(() => {
+    if (!value) return ['', ''];
+    return [Math.floor(value / 60), value % 60];
+  }, [value]);
+
+  const getIntValue = (v?: string | number) => {
+    if (!v) return 0;
+    if (typeof v === 'string') return parseInt(v);
+    return v;
+  };
+
+  const hoursMinutesToValue = memoize(
+    (hours?: string | number, minutes?: string | number) => {
+      const h = getIntValue(hours);
+      const m = getIntValue(minutes);
+      return h * 60 + m;
+    }
+  );
+
+  return (
+    <>
+      {label}
+      <Stack direction='row' gap={1}>
+        <NumberInput
+          {...props}
+          value={hours}
+          onChange={(e) => {
+            if (onChange)
+              onChange(hoursMinutesToValue(e.target.value, minutes));
+          }}
+          label={null}
+          placeholder='Hours'
+          fullWidth={false}
+        />
+        <Typography variant='body1'>:</Typography>
+        <NumberInput
+          {...props}
+          value={minutes}
+          onChange={(e) => {
+            if (onChange) onChange(hoursMinutesToValue(hours, e.target.value));
+          }}
+          label={null}
+          placeholder='Minutes'
```

Comment:
I don't love this as placeholder, because it's not clear that these are minutes/hours if you're viewing a form that's already filled out.

My suggestion would be to make the placeholder `0` for both, and also add an `endAdornment` (a MUI thing) that specifies hours vs minutes. I played around with this, I think it looks decent:
```
          inputWidth={85}
          InputProps={{
            endAdornment: (
              <Typography
                variant='subtitle2'
                color='text.secondary'
                component='span'
              >
                hours
              </Typography>
            ),
            ...props.InputProps, // need to make sure we pass this so parent component can send any other InputProps
          }}

```
<img width="242" alt="Screenshot 2024-02-20 at 4 30 44 PM" src="https://github.com/greenriver/hmis-frontend/assets/5333982/f846db6f-31b9-41e6-b377-f0cf196ef389">


<img width="271" alt="Screenshot 2024-02-20 at 4 31 28 PM" src="https://github.com/greenriver/hmis-frontend/assets/5333982/b4c1d9eb-3799-41f1-b16f-7f09745c0710">

Reply:
This is nice! Thanks for the suggestion. 🎉